src/foamadapter/solver/pimplefoam.py

```diff
@@ -0,0 +1,144 @@
+
+import pybFoam
+from pybFoam import (
+    Info,
+    Time,
+    Word,
+    adjustPhi,
+    computeCFLNumber,
+    constrainHbyA,
+    constrainPressure,
+    createPhi,
+    dictionary,
+    fvc,
+    fvm,
+    fvMesh,
+    fvScalarMatrix,
+    fvVectorMatrix,
+    pimpleControl,
+    setRefCell,
+    solve,
+    surfaceScalarField,
+    volScalarField,
+    volVectorField,
+)
+from pybFoam.turbulence import incompressibleTurbulenceModel, singlePhaseTransportModel
+
+
+class CFLNumber:
+
+    def __init__(self):
+        self.criteria = []
+        self.GREAT = 1e30
+        self.SMALL = 1e-15
+        self.maxDeltaT = 1e-3  # Default maximum deltaT
+
+    def setDelta(self, runTime, phi):
+        deltaT = runTime.deltaTValue()
+
+        if not self.criteria:
+            return
+
+        max_cfl_number, mean_cfl_number = computeCFLNumber(phi)
+        Info(f"Courant Number mean: {mean_cfl_number}, max: {max_cfl_number}")
+        ratios = [self.max_cfl_number / max_cfl_number]
+
+        # limit to 1.2 to avoid too large time steps
+        ratios = [min(ratio, 1.2) for ratio in ratios if ratio > self.SMALL and ratio < self.GREAT]
+
+        # Set most restrictive time step
+        finalDeltaT = min(min(deltaT * ratio for ratio in ratios), self.maxDeltaT)
+        runTime.setDeltaT(finalDeltaT)
+        Info(f"deltaT = {runTime.deltaTValue()}")
+        runTime.increment()
```

Comment:
Should that be part of the solver? I think it should live in the implementation part the bindings.